examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithJson.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.examples.snappydata
+
+import scala.util.Try
+
+import com.typesafe.config.{Config, ConfigFactory}
+import org.apache.log4j.{Level, Logger}
+
+import org.apache.spark.sql.{SnappyContext, SnappyJobInvalid, SnappyJobValid, SnappyJobValidation, SnappySQLJob, SnappySession, SparkSession}
+
+/**
+ * This is a sample code snippet to work with JSON files and SnappyStore column tables.
+ * Run with
+ * <pre>
+ * bin/run-example snappydata.WorkingWithJson quickstart/src/main/resources
+ * </pre>
+ * Also you can run this example by submitting as a job.
+ * <pre>
+ *   cd $SNAPPY_HOME
+ *   bin/snappy-job.sh submit
+ *   --app-name JsonApp
+ *   --class org.apache.spark.examples.snappydata.WorkingWithJson
+ *   --app-jar examples/jars/quickstart.jar
+ *   --lead [leadHost:port]
+ *   --conf json_resource_folder=../../quickstart/src/main/resources
+ *
+ * Check the status of your job id
+ * bin/snappy-job.sh status --lead [leadHost:port] --job-id [job-id]
+ */
+object WorkingWithJson extends SnappySQLJob {
+
+  private val NPARAMS = 1
+
+  private var jsonFolder: String = ""
+
+  override def isValidJob(sc: SnappyContext, config: Config): SnappyJobValidation ={
+    {
+      Try(config.getString("json_resource_folder"))
+          .map(x => SnappyJobValid())
+          .getOrElse(SnappyJobInvalid("No json_resource_folder config param"))
+    }
+  }
+
+  override def runSnappyJob(sc: SnappyContext, jobConfig: Config): Any = {
+
+    val some_people_path = s"${jobConfig.getString("json_resource_folder")}/some_people.json"
+    // Read a JSON file using Spark API
+    val people = sc.jsonFile(some_people_path)
+    people.printSchema()
+
+    //Drop the table if it exists.
+    sc.dropTable("people", ifExists = true)
+
+    // Write the created DataFrame to a column table.
+    people.write.format("column").saveAsTable("people")
+
+    // Append more people to the column table
+    val more_people_path = s"${jobConfig.getString("json_resource_folder")}/more_people.json"
+
+    //Explicitly passing schema to handle record level field mismatch
```

Comment:
One of the most important explorations to me is the support for sparse JSON or objects. Afaik, this is possible with spark. The schema is inferred by looking into the entire batch passed in. You should explore/talk about this. Super important.  Essentially, you can claim to truly be "Not only SQL"